core/src/main/java/tc/oc/pgm/rotation/vote/VotePoolOptions.java

```diff
@@ -32,32 +35,42 @@ public boolean toggleMode() {
     return replace;
   }
 
+  public boolean canAddVote() {
+    return customVoteMaps.size() < MapVotePicker.MAX_VOTE_OPTIONS;
+  }
+
   public boolean addVote(MapInfo map) {
-    if (customVoteMaps.size() < MapVotePicker.MAX_VOTE_OPTIONS) {
-      this.customVoteMaps.add(map);
+    return addVote(map, null);
+  }
+
+  public boolean addVote(MapInfo map, @Nullable UUID playerId) {
+    if (canAddVote()) {
+      this.customVoteMaps.put(map, null);
       return true;
     }
     return false;
   }
 
   public boolean removeMap(MapInfo map) {
-    return this.customVoteMaps.remove(map);
+    boolean present = this.customVoteMaps.containsKey(map);
+    this.customVoteMaps.remove(map);
+    return present;
   }
 
   public Set<MapInfo> getCustomVoteMaps() {
-    return customVoteMaps;
+    return ImmutableSet.copyOf(customVoteMaps.keySet());
```

Comment:
you can do with an unmodifiableSet that wraps the keys, no need to copy this i believe